src/server/controllers/sendChangePasswordEmail.ts

```diff
@@ -173,28 +223,77 @@ export const sendEmailInOkta = async (
 					// if the user is PROVISIONED, we need to reactivate them before we can send them an email
 					// this will keep them in the PROVISIONED state
 					// we will send them a create password email
-					const tokenResponse = await reactivateUser(user.id);
-					if (!tokenResponse?.token.length) {
-						throw new OktaError({
-							message: `Okta user reactivation failed: missing re-activation token`,
-						});
-					}
-					const emailIsSent = await sendCreatePasswordEmail({
-						to: user.profile.email,
-						setPasswordToken: await encryptOktaRecoveryToken({
-							token: tokenResponse.token,
-							appClientId,
-							request_id,
-						}),
-						ref,
-						refViewId,
-					});
-					if (!emailIsSent) {
-						trackMetric(emailSendMetric('OktaCreatePassword', 'Failure'));
-						throw new OktaError({
-							message: `Okta user reactivation failed: Failed to send email`,
+					try {
+						const tokenResponse = await reactivateUser(user.id);
+						if (!tokenResponse?.token.length) {
+							throw new OktaError({
+								message: `Okta user reactivation failed: missing re-activation token`,
+							});
+						}
+						const emailIsSent = await sendCreatePasswordEmail({
+							to: user.profile.email,
+							setPasswordToken: await encryptOktaRecoveryToken({
+								token: tokenResponse.token,
+								appClientId,
+								request_id,
+							}),
+							ref,
+							refViewId,
 						});
+						if (!emailIsSent) {
+							trackMetric(emailSendMetric('OktaCreatePassword', 'Failure'));
+							throw new OktaError({
+								message: `Okta user reactivation failed: Failed to send email`,
+							});
+						}
+					} catch (error) {
+						// these users are in a special state where they used a passcode to create their account
+						// but failed to complete the flow by setting a password, leaving them in an unusual state
+						// where they're in the PROVISIONED state but cannot recover normally
+						// to remediate these users we have to deactivate them
+						// and then reactivate them in order to send them a create password email
+						if (error instanceof OktaError && error.code === 'E0000038') {
+							trackMetric(
+								'PasscodePasswordNotCompleteRemediation-ResetPassword-PROVISIONED-Start',
+							);
+
+							// 1. deactivate the user
+							try {
+								await deactivateUser(user.id);
+								trackMetric('OktaDeactivateUser::Success');
+							} catch (error) {
+								trackMetric('OktaDeactivateUser::Failure');
+								logger.error(
+									'Okta user deactivation failed',
+									error instanceof OktaError ? error.message : error,
+									{
+										request_id,
+									},
+								);
+								throw error;
+							}
+
+							trackMetric(
+								'PasscodePasswordNotCompleteRemediation-ResetPassword-PROVISIONED-Complete',
+							);
+
+							// rerun the sendEmailInOkta function to catch the user in the DEPROVISIONED state
+							return sendEmailInOkta(req, res);
```

Comment:
now that I see this whole logic is duplicated down here for the provisioned users I'm not so sure that it's impossible to go into loops activating and deactivating the user, I'm not familiar enough with these Okta calls